### PR TITLE
feat(tooltip): teleport content to avoid stacking issues

### PR DESCRIPTION
**Pull request by brc-dd**
TODO

- [ ] revert story styles, extra tooltips and long message before merging

### CODE REVIEW

File: .env.d.ts
Comment (brc-dd) [inline .env.d.ts:1]: - moved this to `env.d.ts`, along with content from `lib/types/shims.d.ts`
- needed to explicitly specify .d.ts in path there
- needed to explicitly install `@vue/runtime-core` for module augmentation to work (weirdness because of pnpm, other option was to set `shamefully-hoist=true` in `.npmrc`)
- needed to explicitly add it to `tsconfig.includes`
- removed `declare module '*.vue'` as it seem to be no longer needed and it was suppressing TS errors for missing components, wrong paths, typos, difference in case, etc.
- moving shims outside lib shouldn't have any effect on other projects because they already have vite/client referenced in their respective shims

Volar is now picking up Histoire components like HstSelect, HstText, HstNumber, Story, ... 🎉

File: .gitignore
Comment (brc-dd) [inline .gitignore:13]: vite and vitepress stuff

File: docs/.vitepress/config.mts
Comment (brc-dd) [inline docs/.vitepress/config.mts:9]: This uses Netlify's build-time environment variables to get corresponding story URL - i.e. if you now open docs in deploy preview and click on "View in Histoire" button it will open the story deploy preview. The reverse is also same - "View Documentation" button should point to docs' deploy preview if you're viewing deploy preview of story.

File: docs/.vitepress/config.mts
Comment (brc-dd) [inline docs/.vitepress/config.mts:52]: Doing SSR with teleports. I knew someone added it to vitepress, but I didn't know how to even use it. Didn't find any public repo on GitHub using it. So here is a hacky implementation. 🫠

- It extracts teleported html for `#sefirot-modals` and generate the element for it.
- VitePress renders `teleports.body` in `<body>`, so I'm appending the `#sefirot-modals` element so that it ends up at correct place.
- This needs to be added for empty html too so that teleport target is available when VitePress performs SPA routing and gets some element that needs it.

In dev this is added via theme, similar to histoire and test setup.

File: stories/components/STooltip.01_Playground.story.vue
Comment (brc-dd) [inline stories/components/STooltip.01_Playground.story.vue:48]: revert this and styles before merging

File: stories/styles.css
Comment (brc-dd) [inline stories/styles.css:15]: needed for some reason to make scroll event handlers work in histoire

File: tests/components/SInputAddon.spec.ts
Comment (brc-dd) [inline tests/components/SInputAddon.spec.ts:6]: moved to vitest setup

File: tests/components/SInputBase.spec.ts
Comment (brc-dd) [inline tests/components/SInputBase.spec.ts:32]: `find` doesn't work with teleported targets, so using `querySelector` directly. `findComponent` works, but the teleported stuff inside tooltip needs to be a separate component for it to work.

File: tests/components/STooltip.spec.ts
Comment (brc-dd) [inline tests/components/STooltip.spec.ts:31]: these are skipped, can't make them work

File: tests/composables/V.spec.ts
Comment (brc-dd) [inline tests/composables/V.spec.ts:9]: moved to vitest setup

File: vitest.config.mts
Comment (brc-dd) [inline vitest.config.mts:40]: warnings from SLink.spec.ts because of our fake props

File: tests/components/SLink.spec.ts
Comment (brc-dd) [inline tests/components/SLink.spec.ts:31]: these changes are needed because router is now being injected via setup _for all pages_

RouterLink get's rendered so we can no longer find `router-link` as it will rendered to `<a ...>`. Hence using findComponent instead.

Also, this means that the tests testing that `a` gets created will pass in all cases. We need to test that anchor is getting rendered without `RouterLink` when link is or is marked as external.

File: lib/components/STooltip.vue
Comment (brc-dd) [inline lib/components/STooltip.vue:1]: Huh, finally the tooltip part. Too many things here.

- moved from custom implementation to vueuse stuff because it was cleaner
- I have changed the naming (ignore arrow):
   image
   this is to sync with other libraries like ark/zag, radix, ...\
   content now refers to the tooltip content not the trigger content.

details:

- closes on escape (same as earlier)
- doesn't close when selecting stuff in tooltip (done via delayLeave: 100ms)
- if triggered because of focussing the trigger, it respects specified timeout if any
- padding is now added via css itself as handling it in useTooltip was getting complex
- the tooltip is visible after a delay of 200ms to avoid showing when someone is just scrolling past the element having tooltip. doing this via delayEnter was preventing transition to work, so added via setTimeout on on.value in useTooltip.show (hide clears the timeout if cursor is moved before tooltip is shown)
- weird stuff because of `@types/node` leaking because of shared `tsconfig.json`. So using `window.setTimeout` instead of `setTimeout` (node one has different return type).
- `setTimeout` shouldn't return 0, but still keeping the checks as `!= null`.
- hide blurs the trigger or it's descendants on call (same as earlier)
- bunch of position stuff to avoid rendering tooltip off-screen
- in `getCurrentPositions` there are things like `- bodyRect.top` and `- bodyRect.left` because the position is `absolute` (relative to body). If it was fixed (relative to screen) those can be removed.

File: docs/.vitepress/config.mts
Comment (kiaking) [reply docs/.vitepress/config.mts:9]: 👍

**Comment by kiaking** [comment]
@brc-dd 
Tested and working great! Very nice work. I've reverted the story. Feel free to merge when your ready 👍

**Review by kiaking — APPROVED** [review]

**Comment by brc-dd** [comment]
Ah shit. I merged before you pushed your story 😅

**Comment by kiaking** [comment]
lol Yeah I though I pushed but turns out it was blocked by your upstream branch merge 😆

**Pull request merged**
